tools/nodejs_api/src_nodejs/queryResult.js

```diff
@@ -2,47 +2,60 @@ class QueryResult {
     #connection
     #database;
     #queryResult
-    #isClosed
     constructor(connection, queryResult) {
-        if (typeof queryResult !== "object" || queryResult.constructor.name !==  "NodeQueryResult"){
-            throw new Error("QueryResult constructor requires a NodeQueryResult object as an argument");
+        if (typeof connection !== "object" || connection.constructor.name !==  "Connection"){
```

Comment:
Since the user does not call the constructor of `QueryResult` (i.e. the only way to get the object is through the return value of one of our function), there is no need to do this type of check. You should change it to assertion or just don't do a check at all.